### PR TITLE
FEATURE: Allow admins to delete user SSO records in the UI

### PR DESCRIPTION
**Pull request by pmusaraj**
Also displays the user's last payload in the admin UI to help with debugging SSO issues.

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]
Looks great. This is a nice feature. :+1:

File: app/assets/javascripts/admin/models/admin-user.js [hunk 1]
```diff
@@ -567,6 +567,18 @@ const AdminUser = User.extend({
   _formatError(event) {
     return `http: ${event.status} - ${event.body}`;
   },
+
+  deleteSSORecord() {
+    return ajax(`/admin/users/${this.id}/sso_record.json`, {
+      type: "DELETE",
+    })
+      .then(() => {
+        this.setProperties({
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/models/admin-user.js:576]: nit pick but we don't need setProperties here just set

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -498,6 +499,12 @@ def reset_bounce_score
     render json: success_json
   end
 
+  def sso_record
+    guardian.ensure_can_delete_sso_record!(@user)
+    @user.single_sign_on_record.destroy
```
Comment (tgxworld) [inline app/controllers/admin/users_controller.rb:504]:
```suggestion
    @user.single_sign_on_record.destroy!
```

File: lib/guardian/user_guardian.rb [hunk 3]
```diff
@@ -167,4 +167,7 @@ def can_upload_user_card_background?(user)
     (is_me?(user) && user.has_trust_level?(SiteSetting.min_trust_level_to_allow_user_card_background.to_i)) || is_staff?
   end
 
+  def can_delete_sso_record?(user)
+    SiteSetting.enable_sso && user && is_staff?
```
Comment (tgxworld) [inline lib/guardian/user_guardian.rb:171]: Should this be `is_admin?` instead? `is_staff?` will return true for moderators and admins.

File: spec/requests/admin/users_controller_spec.rb [hunk 4]
```diff
@@ -1049,4 +1049,18 @@ def find_logs(action)
     end
   end
 
+  describe '#sso_record' do
+    fab!(:sso_user) { Fabricate(:user) }
```
Comment (tgxworld) [inline spec/requests/admin/users_controller_spec.rb:1053]: minor but we already have `fab!(:user) { Fabricate(:user) }` at the top of this file so we can just use that.

**Pull request merged**